prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -178,13 +199,21 @@ static void client_recv_connresp(struct util_wait *wait,
 	free(cm_ctx);
 	return;
 err:
-	memset(&err_entry, 0, sizeof err_entry);
 	err_entry.fid = cm_ctx->fid;
 	err_entry.context = cm_ctx->fid->context;
 	err_entry.err = -ret;
-	FI_DBG(&tcpx_prov, FI_LOG_EP_CTRL, "fi_eq_write the conn refused %d\n", ret);
-
+	if (cm_ctx->cm_data_sz) {
+		err_entry.err_data = calloc(1, cm_ctx->cm_data_sz);
```

Comment:
fi_eq_write() below can fail with ENOMEM. In that case err_data would not be freed.